src/FSharpVSPowerTools.Logic/FsiReferenceCommand.fs

```diff
@@ -141,29 +136,56 @@ type FsiReferenceCommand(dte2: DTE2, mcs: OleMenuCommandService, _shell: IVsUISh
                             match referenceAssembliesFolder with
                             | Some referenceAssembliesFolder ->
-                                if String.Equals(Path.GetFullPathSafe(referenceAssembliesFolder), 
-                                                 Path.GetFullPathSafe(referenceFolder), StringComparison.OrdinalIgnoreCase) then
-                                    assemblyRefList.Add(Path.GetFileName(fullPath))
+                                if String.Equals(Path.GetFullPathSafe referenceAssembliesFolder, 
+                                                 Path.GetFullPathSafe referenceFolder, StringComparison.OrdinalIgnoreCase) then
+                                    assemblyRefList.Add(Path.GetFileName fullPath)
                                 else
                                     assemblyRefList.Add(getRelativePath scriptFolder fullPath)
                             | _ -> assemblyRefList.Add(getRelativePath scriptFolder fullPath))
         | _ -> ()
 
-        let assemblyRefList = assemblyRefList |> Seq.map (sprintf "#r @\"%s\"") |> Seq.toList
-        let projectRefList = projectRefList |> Seq.map (sprintf "#r @\"%s\"") |> Seq.toList
-        String.Join(Environment.NewLine, 
-                    headerComment,
-                    String.Join(Environment.NewLine, assemblyRefList),
-                    String.Join(Environment.NewLine, projectRefList))
+        assemblyRefList 
+        |> Seq.append projectRefList
+        |> Seq.map (sprintf "#r @\"%s\"") 
+        |> Seq.toList
+
+    let generateFileContent (refLines: #seq<string>) =
+        String.Join(Environment.NewLine, headerComment, String.Join(Environment.NewLine, refLines))
+
+    let tryGetExistingFileRefs project fileName = 
+        let filePath = getFullFilePathInScriptFolder project fileName  
+        if File.Exists filePath then
+            File.ReadLines filePath
+            |> Seq.filter (fun (line: string) -> line.StartsWith "#r")
+            |> Seq.toList
+            |> Some
+        else None
 
     let generateFile (project: Project) =
         Option.ofNull project
         |> Option.iter (fun project ->
-            addFileToActiveProject(project, "load-references.fsx", generateFileContent project)
+            let loadRefsFileName = "load-references.fsx"
+            let actualRefs = generateRefs project
+            let refs = 
+                match tryGetExistingFileRefs project loadRefsFileName with
+                | None -> actualRefs
+                | Some existingRefs ->
+                    // remove refs which don't actual anymore (they have been removed from the project)
+                    let existingRefs =
+                        existingRefs 
+                        |> List.filter (fun existingRef -> List.exists ((=) existingRef) actualRefs)
+                    // get refs which don't exist in the existing file
+                    let newExtraRefs =
```

Comment:
`newExtraRefs` is always empty since `existingRefs` is shadowed and contains only elements from `actualRefs`. So the code doesn't work if there is some added references that don't appear in newly generated script.

Probably remove (evil) value shadowing and calculate `survivedRefs` in the first step.